pkg/container/client.go

```diff
@@ -111,9 +115,59 @@ type Client struct {
 	sysCtx *types.SystemContext
 }
 
+func isImageIdentifier(target string) bool {
+	if ok := reference.ShortIdentifierRegexp.MatchString(target); ok {
+		return true
+	}
+
+	if ok := reference.IdentifierRegexp.MatchString(target); ok {
+		return true
+	}
+
+	return false
+}
+
+// the short identifier has a minimum length of 6
+// returning the full-length identifier causes issues
+// for the named parsing. So we should shorten this.
+func truncateIdentifier(id string) string {
+	if length := len(id); length < 12 {
+		return id[:length]
+	}
+	return id[:12]
+}
+
+func parseImageIdentifier(target string) (reference.Reference, error) {
+	if !isImageIdentifier(target) {
+		return nil, nil
+	}
+
+	id := truncateIdentifier(target)
+	ref, err := reference.Parse(id)
+	if err != nil {
+		return nil, err
+	}
+	return ref, nil
+}
+
 // NewClient constructs a new Client for target with default options.
 // It will add the "latest" tag if target does not contain it.
 func NewClient(target string) (*Client, error) {
+	targetId, err := parseImageIdentifier(target)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse image id '%s': %w", target, err)
+	}
+
+	if targetId != nil {
+		// TODO: the line below is going to create a named
+		// reference of `docker.io/library/{targetid}`.
+		// Some ideas on how to handle this:
+		// 1. Allow `docker.io/library/{targetid}`
+		// 2. Set it to `localhost/unknown`
+		// 3. Set the name to `<none>` which causes an error on the
+		//    `ParsedNormalizedNamed` function call
+		target = UnknownContainerName
```

Comment:
This is pretty janky... but we rely heavily on named references and I didn't want to break that.
So this was a work around.